### PR TITLE
Add Harbor rules

### PR DESCRIPTION
**Pull request by William-Young-97**
Added harbor.rules.yml file with 3 alerts (InstanceDown, HarborUnhealthy and  ProjectQuotaIsRaisingTheLimit).

### CODE REVIEW

**Comment by hervenicol** [comment]
Build fails because there were no unit tests written for this new rules.
See https://github.com/giantswarm/prometheus-rules#testing for how to write them.

Hint: to go faster when running the tests, you can do `make test test_filter=harbor`

File: helm/prometheus-rules/templates/alerting-rules/harbor.all.rules.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        severity: critical
+        app: harbor
+    - alert: ProjectQuotaIsRaisingTheLimit
+      annotations:
+        summary: The quota of the repository '{{` {{ $labels.repo_name }} is getting close to the limit. `}}'
+      expr: harbor_quotas_size_bytes{type="used"} / ignoring(type) harbor_quotas_size_bytes{type="hard"} > 0.95
+      for: 5m
+      labels:
+        severity: critical
+        app: harbor
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/harbor.all.rules.yml:43]: Newline please

File: test/tests/providers/global/harbor.all.rules.test.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (MarcelMue) [inline test/tests/providers/global/harbor.all.rules.test.yml:1]: Not needed I think

File: test/tests/providers/global/harbor.all.rules.test.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+              app: harbor
+              cancel_if_apiserver_down: "true"
+              cancel_if_cluster_status_creating: "true"
+              cancel_if_cluster_status_deleting: "true"
+              cancel_if_cluster_status_updating: "true"
+              cancel_if_scrape_timeout: "true"
+              cancel_if_outside_working_hours: "true"
+            exp_annotations:
+              description: " of job  has been down for more than 5 minutes."
+              summary: " Instance  down."
```
Comment (MarcelMue) [inline test/tests/providers/global/harbor.all.rules.test.yml:24]: Newline

File: helm/prometheus-rules/templates/alerting-rules/harbor.all.rules.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        cancel_if_outside_working_hours: "true"
+    - alert: HarborUnhealthy
+      annotations:
+        description: '{{` {{ $labels.instance }} is unhealthy.`}}'
+        summary: "Habor cluster is unhealthy."
+      expr: harbor_health == 0
+      for: 5m
+      labels:
+        severity: critical
+        app: harbor
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/harbor.all.rules.yml:35]: Should these also cancel on the conditions above?

File: test/tests/providers/global/harbor.all.rules.test.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+rule_files:
+  - harbor.all.rules.yml
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'harbor_up{}'
+        values: "_x5 1+0x5 0+0x5"
+    alert_rule_test:
+      - alertname: HarborDown
+        eval_time: 5m
```
Comment (hervenicol) [inline test/tests/providers/global/harbor.all.rules.test.yml:10]: Your values say:
* from t0 to t4, no data
* from t5 to t9, "harbor_up=1"
* from t10 to t14 "harbor_down=0"

At `t5`, do you really expect an alert to fire?
I'm not sure data is added before or after the test, but either you have no data (ie probably harbor is not installed) or you have just one data point saying it's running.

This means, on clusters where harbor is not installed this alert will fire?

**Comment by William-Young-97** [comment]
@MarcelMue Should be good to go now pending review.

File: test/tests/providers/global/harbor.all.rules.test.yml (hunk 5 above)
Comment (William-Young-97) [reply test/tests/providers/global/harbor.all.rules.test.yml]: Would it be a good idea to have these unit tests as separate series and values?

**Comment by William-Young-97** [comment]
@hervenicol Something like this?

**Comment by hervenicol** [comment]
> @hervenicol Something like this?

Yes, looking good for `HarborDown` tests! :+1: 
Could you apply the same logic to other alerts tests as well?

**Comment by hervenicol** [comment]
The PR looks good from a technical point of view.

Now it's missing a few things as per https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured:

- [ ] `CHANGELOG.md` update
- [ ] `opsrecipe` annotation
- [ ] `dashboard` annotation (if you have some Grafana dashboard for Harbor)
- [ ] `area` label
- [ ] `team` label

...but it may be more @MarcelMue 's job for some.

**Comment by MarcelMue** [comment]
Needs to be finalized by @giantswarm/team-honeybadger before merge.

**Comment by QuentinBisson** [comment]
@MarcelMue do we still need to keep this PR?

**Pull request closed without merging**